front-end/src/Views/ProjectPage.js

```diff
@@ -322,6 +341,41 @@ export default class HomePage extends Component {
         }); 
   }
 
+  renderColabRequests() {
+    const { allColabRequests } = this.state;
+    var userIds = [];
+    for(var i=0; i<allColabRequests.length; i++){
+      userIds.push(allColabRequests[i].from_user);
+    }
+    var uniqueIds = userIds.filter(this.distinct);
+  
+    if (uniqueIds.length === 0) return (  <Typography variant='h6' color="textPrimary">No Collaborators</Typography>)
+    else return uniqueIds.map((item) => {
+      return (<>
+        <Typography variant="h6" color="primary" style={{ cursor: "pointer", width: "100%", textAlign: "center" }}>{this.getProfileNameById(item)}</Typography>

```

Comment:
I think, you can get profile name with getProject() function via member data. Also, in a function, data is got with api call, cannot be returned and the function returns empty or null.

Reply:
It's better to get the data in ComponentDidMount or somewhere else, (when a button is clicked get this data etc. ) 
Set it as a state variable and use it in rendering.